noether_examples/src/mesh_segmenter_client.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2016, Southwest Research Institute
+ * All rights reserved.
+ *
+ */
+
+#include <ros/ros.h>
+
+#include <gtest/gtest.h>
+#include <pcl/PCLPointCloud2.h>
+#include <pcl/point_traits.h>
+#include <pcl/point_types.h>
+#include <pcl/common/io.h>
+#include <pcl/console/print.h>
+#include <pcl/io/ply_io.h>
+#include <pcl/io/ascii_io.h>
+#include <pcl/io/vtk_lib_io.h>
+#include <fstream>
+#include <locale>
+#include <stdexcept>
+
+#include <pcl/surface/vtk_smoothing/vtk_utils.h>
+#include <vtkSmartPointer.h>
+#include <vtkPolyData.h>
+
+#include <noether_msgs/SegmentAction.h>
+#include <actionlib/client/service_client.h>
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <noether/noether.h>
+#include <vtkSTLWriter.h>
+#include <ros/package.h>
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "noether_node");
+  ros::NodeHandle pnh("~");
+
+  // Step 1: Load parameters
+  std::string file;
+  double curvature_threshold;
+  int min_cluster_size, max_cluster_size;
+  bool show_individually, save_outputs;
+  pnh.param<std::string>("filename", file, "");
+  pnh.param<int>("min_cluster_size", min_cluster_size, 500);
+  pnh.param<int>("max_cluster_size", max_cluster_size, 1000000);
+  pnh.param<double>("curvature_threshold", curvature_threshold, 0.3);
+  pnh.param<bool>("show_individually", show_individually, false);
+  pnh.param<bool>("save_outputs", save_outputs, false);
+
+  // Step 2: Import the mesh
+  pcl::PolygonMesh pcl_mesh;
+  pcl::io::loadPLYFile(file, pcl_mesh);
+  std::cout << "Imported as PCL mesh of size " << pcl_mesh.cloud.data.size() << '\n';
```

Comment:
Once again, you're using cout instead of ROS_INFO, ROS_ERROR etc. Why?

Reply:
No good reason.